File: 2.Algorithm/01.Brute-force/소수 찾기/소수 찾기.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

class Solution {

    private static int isPrime(int n) {
    	if (n < 2) {return 0;}
    	for(int j = 2; j < (int)Math.sqrt(n)+1; j++){
            if (n % j == 0){
                return 0;
            }
        }
        return 1;
    }
    
	private static void permutation( String[] numbers, boolean[] isChecked, int dept, String[] result,  Set<Integer> totalList ) {
	    for ( int i = 0; i < numbers.length; i++ ) {
	        if( !isChecked[i] ) {
	            isChecked[i] = true;
	            result[dept] = numbers[i];
	            totalList.add(Integer.parseInt(String.join("", result)));
	            permutation(numbers, isChecked, dept + 1, result, totalList);
	            isChecked[i] = false; 
	            result[dept] = ""; 
	            }
	    }
	}
    
    public int solution(String number) {
		String[] numbers = number.split("");
        boolean[] isChecked = new boolean[numbers.length];
	    Set<Integer> totalList = new HashSet<Integer>();
	    String[] result = new String[numbers.length];
	    Arrays.fill(result, "");
	    permutation(numbers, isChecked, 0, result, totalList);
	    int answer = 0;
	    for (int i : totalList) {
	        answer += isPrime(i);
	    }
        return answer;
    }
}

File: 2.Algorithm/01.Brute-force/소수 찾기.py
import itertools

def solution(numbers):
    a = []
    for i in range(0,len(numbers)):
        a += list(map(int,map(''.join, itertools.permutations(numbers, i+1))))
    a = sorted(list(set(a)))
    answer = len(a)
    for i in a:
        for j in range(2,int(i**0.5)+1):
            if i % j == 0:
                answer -=1
                break
    return answer - sum([1 if x == 1 or x == 0 else 0 for x in a])